.github/workflows/test.yml

```diff
@@ -5,6 +5,11 @@ on: [push, pull_request]
 permissions:
   contents: read
 
+# Cancel ongoing builds on new changes
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event_name }}-${{ github.head_ref || github.ref }}
```

Comment:
I'm surprised GitHub does not provide a shorter, more user friendly way to express this.

Reply:
It is [documented](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/control-the-concurrency-of-workflows-and-jobs#example-using-concurrency-and-the-default-behavior) like that, but I copied this from the Zephyr project workflows. Adding more specific IDs to the `group` will make sure we only cancel our PR's runs.